packages/opentelemetry-resources/src/Resource.ts

```diff
@@ -25,6 +26,16 @@ import { defaultServiceName } from './platform';
  */
 export class Resource {
   static readonly EMPTY = new Resource({});
+  private _syncAttributes: ResourceAttributes;
+  private _asyncAttributesPromise: Promise<ResourceAttributes> | undefined;
```

Comment:
I think adding these private properties to `Resource` will make it so this class will not be compatible between different core versions, similar to the problems we are facing with `Span` when using a concrete class and not interface in public API.

Once we merge this PR there is no way back. Do you think it is still possible to create a `Resource` interface so we don't run into issues? @Flarna